packages/web-fragments/src/gateway/middleware/web.ts

```diff
@@ -144,6 +145,7 @@ export function getWebMiddleware(
 		 */
 		const fragmentAssetResponse = new Response(fragmentResponse.body, fragmentResponse);
 		fragmentAssetResponse.headers.append('x-web-fragment-id', matchedFragment.fragmentId);
+		enforceIdentityEncoding(fragmentAssetResponse);
```

Comment:
could you please add some tests for this in gateway spec? thanks

it's a bit odd that we need to do this, it's as if something upstream decoded the stream but didn't update the headers. do we know who did that? htmlrewriter? something else?